src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -194,7 +201,7 @@ private ISaslStep Transition(SaslConversation conversation, ISaslStep currentSte
         /// <summary>
         /// Represents a SASL conversation.
         /// </summary>
-        protected sealed class SaslConversation : IDisposable
+        protected internal sealed class SaslConversation : IDisposable
```

Comment:
Since `ISaslStep` is now `internal` and ISaslStep's  has an `Initialize` method takes in a `SaslConversation`, `SaslConversation` can no longer simply be `protected`: it needs to be promoted to `internal` as well.

Reply:
Would not be necessary if you adopt my suggestion above.